worker/uniter/remotestate/watcher.go

```diff
@@ -208,7 +208,7 @@ func (w *RemoteStateWatcher) loop(unitTag names.UnitTag) (err error) {
 	requiredEvents++
 
 	var seenRelationsChange bool
-	relationsw, err := w.service.WatchRelations()
```

Comment:
There's updates needed to the unit tests.
Currently, there's a relationsWatcher on mockService but things will need to be reworked so that relationsWatcher is on mockUnit; and the tests will need to be enhanced to check that only the correct units get events etc